src/ImageSharp/Configuration.cs

```diff
@@ -1,267 +1,221 @@
+        }
+
+        /// <summary>
+        /// Sets a specific image encoder as the encoder for a specific image format.
+        /// </summary>
+        /// <param name="imageFormat">The image format to register the encoder for.</param>
+        /// <param name="encoder">The encoder to use,</param>
+        public void SetEncoder(IImageFormat imageFormat, IImageEncoder encoder)
+        {
+            this.FormatsManager.SetEncoder(imageFormat, encoder);
+        }
+
+        /// <summary>
+        /// Sets a specific image decoder as the decoder for a specific image format.
+        /// </summary>
+        /// <param name="imageFormat">The image format to register the encoder for.</param>
+        /// <param name="decoder">The decoder to use,</param>
+        public void SetDecoder(IImageFormat imageFormat, IImageDecoder decoder)
+        {
+            this.FormatsManager.SetDecoder(imageFormat, decoder);
+        }
+
+        /// <summary>
+        /// Removes all the registered image format detectors.
+        /// </summary>
+        public void ClearImageFormatDetectors()
+        {
+            this.FormatsManager.ClearImageFormatDetectors();
+        }
+
+        /// <summary>
+        /// Adds a new detector for detecting mime types.
+        /// </summary>
+        /// <param name="detector">The detector to add</param>
+        public void AddImageFormatDetector(IImageFormatDetector detector)
+        {
+            this.FormatsManager.AddImageFormatDetector(detector);
+        }
+
+        /// <summary>
+        /// For the specified mime type find the decoder.
+        /// </summary>
+        /// <param name="format">The format to discover</param>
+        /// <returns>The <see cref="IImageDecoder"/> if found otherwise null</returns>
+        public IImageDecoder FindDecoder(IImageFormat format)
+        {
+            return this.FormatsManager.FindDecoder(format);
+        }
+
+        /// <summary>
+        /// For the specified mime type find the encoder.
+        /// </summary>
+        /// <param name="format">The format to discover</param>
+        /// <returns>The <see cref="IImageEncoder"/> if found otherwise null</returns>
+        public IImageEncoder FindEncoder(IImageFormat format)
+        {
+            return this.FormatsManager.FindEncoder(format);
+        }
+
+        /// <summary>
+        /// Creates the default instance with the following <see cref="IConfigurationModule"/>s preregistered:
+        /// <para><see cref="PngConfigurationModule"/></para>
+        /// <para><see cref="JpegConfigurationModule"/></para>
+        /// <para><see cref="GifConfigurationModule"/></para>
+        /// <para><see cref="BmpConfigurationModule"/></para>
+        /// </summary>
+        /// <returns>The default configuration of <see cref="Configuration"/></returns>
+        internal static Configuration CreateDefaultInstance()
+        {
+            return new Configuration(

```

Comment:
Shouldn't the `ImageFormatManager` constructor accept these instead?

Reply:
That's an issue I would like to discuss. The ´IConfigurationModule´ requires to pass a ´Configuration´ instance. If we want to do the formats configuration in 'ImageFormatManager'  the parameter needs to be changed from Configuration to 'ImageFormatManager', which would lead to renaming the interface to 'IImageFormatModule' or something like that.

But I can see the ´IConfigurationModule´ to be useful to intialize things other than image formats, so I believe it's still subject to debate.

Reply:
nope they should be passed into the `Configuration` like they are and not into `ImageFormatManager` manager its just happen by chance that the things they are configuring are image formats rather than anything else.

Reply:
Great! keep it as-is then